text/src/tr.rs

```diff
@@ -298,38 +381,38 @@ fn parse_classes(input: &str) -> Result<(Vec<Operand>, CaseSensitive), String> {
     while let Some(&ch) = chars.peek() {
         if ch == '[' {
             chars.next(); // Skip '['
-            if let Some(&':') = chars.peek() {
-                // Processing the [:class:] format
-                chars.next(); // Skip ':'
-                let mut class = String::new();
-                while let Some(&next_ch) = chars.peek() {
-                    if next_ch != ':' {
-                        class.push(next_ch);
-                        chars.next();
-                    } else {
-                        break;
-                    }
-                }
-                if class.is_empty() {
-                    return Err("Error: Missing class name after '[:'".to_string());
-                }
-                if let Some(&':') = chars.peek() {
-                    chars.next(); // Skip ':'
-                    if let Some(&']') = chars.peek() {
-                        chars.next(); // Skip ']'
-                        let res = expand_character_class(&class)?;
-                        case_sensitive = res.1;
-                        classes.extend(res.0);
-                    } else {
-                        return Err("Error: Missing closing ']' for '[:class:]'".to_string());
-                    }
-                } else {
-                    return Err("Error: Missing ':' before ']' for '[:class:]'".to_string());
-                }
-            } else {
+
+            let Some(&':') = chars.peek() else {
```

Comment:
Both cases use next right after peek. Can't we just `.next` on `let-else` pattern?